mobile/examples/whisper/android/azure/app/src/main/java/ai/onnxruntime/example/speechrecognition/AudioTensorSource.kt

```diff
@@ -0,0 +1,103 @@
+package ai.onnxruntime.example.speechrecognition
+
+import ai.onnxruntime.OnnxJavaType
+import ai.onnxruntime.OnnxTensor
+import ai.onnxruntime.OrtEnvironment
+import android.annotation.SuppressLint
+import android.media.AudioFormat
+import android.media.AudioRecord
+import android.media.MediaRecorder
+import android.util.Log
+import java.nio.ByteBuffer
+import java.nio.ByteOrder
+import java.util.concurrent.atomic.AtomicBoolean
+
+class AudioTensorSource {
+    companion object {
+        private const val sampleRate = 16000
+        private const val maxAudioLengthInSeconds = 30
+
+        fun fromRawWavBytes(rawBytes: ByteArray): OnnxTensor {
```

Comment:
is the model input expected to be a wav file or raw PCM data? from this method name it seems like it is a wav file, but the `fromRecording` method seems to be returning a tensor of PCM data.

Reply:
I believe the model expects a wav as input as the `audio_format` attribute specifies that.

The OpenAI endpoint seems to be flexible though and is able to determine the audio type ignoring what you said it was. Possibly it uses that info as a hint only. Despite that it strangely requires you to provide a filename with an accepted input format as the extension.

It would be better to send in a wav as the Azure OAI endpoint may not be as forgiving and it's confusing for a new user to make sense of the input when it doesn't match what the model appears to require.

Reply:
thanks. - got the `fromRecording` method updated to work with a wav format file too.